apps/allocations/app/components/App/AllocationsHistory.js

```diff
@@ -63,30 +62,34 @@ const AllocationsHistory = ({ allocations }) => {
           </Amount>
         ]
       }}
-      //renderEntryExpansion={({ recipients, amount, token }) => {
-      //  return recipients.map((recipient, index) => {
-      //    const allocated = BigNumber(recipient.amount).div(amount)
-      //    return (
-      //      <Recipient key={index}>
-      //        <IdentityBadge
-      //          entity={recipient.address}
-      //          shorten={true}
-      //        />
-      //        <RecipientProgress>
-      //          <ProgressBar
-      //            value={allocated.toNumber()}
-      //            color={String(theme.accentEnd)}
-      //          />
-      //        </RecipientProgress>
-      //        <RecipientAmount theme={theme}>
-      //          { displayCurrency(BigNumber(recipient.amount)) } {' '}
-      //          {token} {' • '}
-      //          { allocated.times(100).dp(0).toNumber() }{'%'}
-      //        </RecipientAmount>
-      //      </Recipient>
-      //    )
-      //  })
-      //}}
+      renderEntryExpansion={({ recipients, amount, token }) => {
+        const totalSupports = recipients.reduce((total, recipient) => {
+          return total + Number(recipient.supports)
+        }, 0)
+        return recipients.map((recipient, index) => {
+          const allocated = BigNumber(recipient.supports).div(totalSupports)
+          return (
+            <Recipient key={index}>
+              <IdentityBadge
+                entity={recipient.candidateAddress}
+              />
+              <RecipientAllocation>
+                <RecipientProgress>
+                  <ProgressBar
+                    value={allocated.toNumber()}
+                    color={String(theme.accentEnd)}
+                  />
+                </RecipientProgress>
+                <RecipientAmount theme={theme}>
+                  { displayCurrency(BigNumber(amount).times(allocated)) } {' '}
+                  {getTokenSymbol(token)} {' • '}
+                  { allocated.times(100).dp(0).toNumber() }{'%'}
```

Comment:
Same here about logic in the render return

Reply:
Same question here 😄 

"Avoid all logic in render function" seems like an idea from Angular or Bootstrap or something. Avoid _side effects_ in the render function? Yes, absolutely. But logic? This seems like the most natural place for it to live to me, and I'm not even sure if or how we could move it elsewhere.